### PR TITLE
seqfu: new package

### PR DESCRIPTION
**Pull request by dialvarezs**
Seqfu (https://github.com/telatin/seqfu2) is a set of utilities for FASTX files, written in nim.

### CODE REVIEW

**Comment by dialvarezs** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/seqfu/package.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    url = "https://github.com/telatin/seqfu2/archive/refs/tags/v1.20.3.tar.gz"
+
+    license("GPL-3.0", checked_by="dialvarezs")
+
+    version("1.20.3", sha256="1b287b99f3f1ac7045f4d551e781d6780ce168ba8e0a7bfaa0f5490f32e15938")
+
+    depends_on("nim@2", type="build")
+    depends_on("zlib", type="build")
+
+    patch("wno_incompatible_pointer_types.patch", when="%gcc@14:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/seqfu/package.py:22]: Any chance of getting this upstreamed to the repository?

**Review by tldahlgren — APPROVED** [review]
Confirmed the homepage and version sha256.  Have a question about the patch.

File: var/spack/repos/builtin/packages/seqfu/package.py (hunk 1 above)
Comment (dialvarezs) [reply var/spack/repos/builtin/packages/seqfu/package.py:22]: I can open a pr in the repository and see

**Pull request merged**